sample.py

```diff
@@ -0,0 +1,58 @@
+# This function adds two numbers
+def add(x, y):
+    return x + y
+
+
+# This function subtracts two numbers
+def subtract(x, y):
+    return x - y
+
+
+# This function multiplies two numbers
+def multiply(x, y):
+    return x * y
+
+
+# This function divides two numbers
+def divide(x, y):
+    return x / y
+
+
+print("Select operation.")
+print("1.Add")
+print("2.Subtract")
+print("3.Multiply")
+print("4.Divide")
+
+while True:
+    # take input from the user
+    choice = input("Enter choice(1/2/3/4): ")
+
+    # check if choice is one of the four options
+    if choice in ('1', '2', '3', '4'):
+        try:
+            num1 = float(input("Enter first number: "))
+            num2 = float(input("Enter second number: "))
+        except ValueError:
+            print("Invalid input. Please enter a number.")
+            continue
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider extracting input validation logic</b></div>
<div id="fix">

Consider consolidating the error handling for numeric input validation into a separate helper function to improve code reusability and maintainability.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -29,12 +29,16 @@
 +def get_number_input(prompt):
 +    try:
 +        return float(input(prompt))
 +    except ValueError:
 +        raise ValueError("Invalid input. Please enter a number.")
 +
  while True:
      if choice in ('1', '2', '3', '4'):
          try:
 -            num1 = float(input("Enter first number: "))
 -            num2 = float(input("Enter second number: "))
 +            num1 = get_number_input("Enter first number: ")
 +            num2 = get_number_input("Enter second number: ")
          except ValueError:
              print("Invalid input. Please enter a number.")
              continue

```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Amruta101998/DS-and-Algorithm/pull/22#issuecomment-2494167449>#2b78c5</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged